### ISSUE TITLE
Some Feedbin statuses not syncing

### ISSUE THREAD

**Issue by brentsimmons** [Ranchero-Software/NetNewsWire#2715]
It‘s Jan 7, and I just ran NetNewsWire 6 on one of my Macs for the first time in a few days.

On my other Mac and on my iPhone, everything in my Feedbin account is marked as read.

On this Mac, however, I have 27 unread articles. Almost all of them date from Jan. 4. (Those that don’t date from Jan. 3.)

**Comment by vincode-io** [comment]
How old is the build you are using?  Is it possible you don't have this fix in it?  See #2659. 

https://github.com/Ranchero-Software/NetNewsWire/commit/ea407fd8da9a3356bf592acdbc3cb0d9a6b107db

**Comment by brentsimmons** [comment]
New build — run from Xcode, with latest changes.

I’ve gotten both sides caught up. Let’s try this again and see what happens.

**Comment by vincode-io** [comment]
It looks like ever time this gets reported, the person hasn't sync'd for an extended period of time.  I thought I had this cleared up in #2659, but if you had that fix in and it happened again, something else must be going on.

**Comment by vincode-io** [comment]
Possibly related: #2785

**Comment by vincode-io** [comment]
I have been trying to figure this out. The problem is that I can't recreate it and the code looks correct to me. I'm stumped for now.

**Comment by vincode-io** [comment]
So I did finally manage to recreate this. The Feedbin API was returning 304 Not Modified even though the data was changed on the server. Then a short time later it actually gave me a 200 return code.

@benubois Is it possible that there is a lag in updating the conditional get info server side sometimes?

**Comment by benubois** [comment]
Looking into this. What API endpoint are you referring to? /unread_entries.json?

**Comment by vincode-io** [comment]
`unread_entries.json` is the one!

**Comment by benubois** [comment]
Ok,  I've tried two things.

1. Marking things as read on feedbin.com and then checking the etag header from `https://api.feedbin.com/v2/unread_entries.json`. The etag header does change and it changes immediately.

2. To simulate what I imagine NetNewsWire is doing I tried these three `curl` requests. This is a live account so you should be able to run these requests too.

```bash
username="netnewswire-api-test"
password="password"

curl                                           \
--user "${username}:${password}"               \
--verbose                                      \
https://api.feedbin.com/v2/unread_entries.json
```

From the response I copied out the `etag` value and one of the `entry_ids`. Then made a request to mark that entry_id as read:

```bash
curl                                           \
--user "${username}:${password}"               \
--verbose                                      \
--request DELETE                               \
--data '{"unread_entries":[2439174858]}'       \
--header "Content-Type: application/json"      \
https://api.feedbin.com/v2/unread_entries.json

```

Finally, I'm requesting the array of unread_entries again, this time with an `If-None-Match` header set to the `etag` value from the first request:

```bash
curl                                                              \
--verbose                                                         \
--user "${username}:${password}"                                  \
--header 'If-None-Match: W/"6be6ab9650b5721daf04b974f42a19b1"'    \
https://api.feedbin.com/v2/unread_entries.json
```

This returns a `200 OK` and the json payload as expected. When skipping request two and re-requesting `unread_entries.json` with `If-None-Match` it returns a 304 as expected.

I'd be happy to try running NetNewsWire through Proxyman to see if there's something different about the requests that's trigger a 304. Is this just happening in NetNewsWire 6 or the stable version as well?

**Comment by benubois** [comment]
So far the HTTP traffic from NetNewsWire 6.0a2 looks good. The only thing I noticed was that on first launch `unread_entries.json` gets requested twice a few hundred milliseconds apart. 

The days between launches part of this does make it tricky, but I'll try again after not launching for a while to see if the behavior is any different.

<img width="1520" alt="Screen Shot 2021-02-09 at 10 50 00 AM" src="https://user-images.githubusercontent.com/133809/107412946-0a5e8400-6ac5-11eb-88ae-5625fca9535a.png">

**Comment by vincode-io** [comment]
I haven't been able to successfully recreate what I saw earlier today and am beginning to doubt myself. I was testing using my phone and desktop. When I marked the items as read on my phone, I may not have sent it to the background and triggered the sync. That would have meant that my changes didn't make it to the server until background refresh was triggered.

@benubois I may have wasted your time. My apologies and I wouldn't look into this anymore if I were you. I'll keep investigating and let you know if I find interesting.
